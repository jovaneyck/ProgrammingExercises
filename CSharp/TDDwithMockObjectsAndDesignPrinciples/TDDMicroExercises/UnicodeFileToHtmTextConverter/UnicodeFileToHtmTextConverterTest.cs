using System;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace TDDMicroExercises.UnicodeFileToHtmTextConverter
{
    [TestFixture]
    class UnicodeFileToHtmTextConverterTest
    {
        [Test]
        [UseReporter(typeof(NUnitReporter))]
        public void ConvertsAFileToHtml()
        {
            //This test hits the filesystem AND needs a real file for every test run :(
            //Hardcoded dependency on the File API is a violiation of SoliD:
            //Multiple repsonsibilities: reading a file from disk AND converting it to HTML
            //Hardwired dependency on a concrete type, rather than an abstraction: File API.
            const string fullFileName = "./UnicodeFileToHtmTextConverter/testFile.txt";
            UnicodeFileToHtmTextConverter converter = new UnicodeFileToHtmTextConverter(fullFileName);
            string result = converter.ConvertToHtml();
           Approvals.Verify(result);
        }
    }
}
