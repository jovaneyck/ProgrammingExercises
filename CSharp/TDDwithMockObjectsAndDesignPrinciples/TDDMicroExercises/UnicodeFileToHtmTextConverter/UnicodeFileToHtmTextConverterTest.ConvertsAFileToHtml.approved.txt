I am a test file that needs to be &#39;encoded&#39; #in &lt;html&gt;<br />containing two lines!<br />